components/notion/actions/retrieve-block/retrieve-block.mjs

```diff
@@ -15,33 +15,50 @@ export default {
       ],
     },
     retrieveChildren: {
-      type: "boolean",
-      label: "Retrieve Content (Child Blocks)",
-      description: "Retrieve all the children (recursively) for the specified page. [See the documentation](https://developers.notion.com/reference/get-block-children) for more information",
+      type: "string",
+      label: "Retrieve Children",
+      description: "Retrieve all the children (recursively) for the specified page, or optionally filter to include only sub-pages in the result. [See the documentation](https://developers.notion.com/reference/get-block-children) for more information",
       optional: true,
-      default: false,
+      options: [
+        "All Children",
+        "Sub-Pages Only",
+        "None",
+      ],
     },
     retrieveMarkdown: {
       type: "boolean",
       label: "Retrieve as Markdown",
-      description: "Return the page content as markdown instead of block objects.",
+      description: "Additionally return the page content as markdown",
       optional: true,
     },
   },
   async run({ $ }) {
+    let markdownContent;
     if (this.retrieveMarkdown) {
-      const response = await this.notion.getPageAsMarkdown(this.blockId);
-      $.export("$summary", "Successfully retrieved page as markdown");
-      return response;
+      markdownContent = await this.notion.getPageAsMarkdown(this.blockId);
     }
 
+    const { retrieveChildren } = this;
+    const subpagesOnly = retrieveChildren === "Sub-Pages Only";
+
     const block = await this.notion.retrieveBlock(this.blockId);
-    if (this.retrieveChildren) {
-      block.children = await this.notion.retrieveBlockChildren(block);
+    if ([
+      true,
+      "All Children",
+      "Sub-Pages Only",
+    ].includes(retrieveChildren)) {
+      block.children = await this.notion.retrieveBlockChildren(block, subpagesOnly);
     }
-    $.export("$summary", `Successfully retrieved block${this.retrieveChildren
-      ? ` with ${block.children.length ?? 0} children`
+    $.export("$summary", `Successfully retrieved block${retrieveChildren
```

Comment:
This line gets an error when `retrieveChildren` is set to `None`. Should add `&& retrieveChildren !== "None"`:

`$.export("$summary", `Successfully retrieved block${retrieveChildren && retrieveChildren !== "None"`